src/AdditionalPylonsModule.cpp

```diff
@@ -80,7 +80,13 @@ void AdditionalPylonsModule::onUnitCreate(BWAPI::Unit unit) {
 		player.onUnitCreate(unit);
 	}
 	else if(unit->getPlayer() != BWAPI::Broodwar->neutral()){
-		enemy.onUnitCreate(unit);
+		if (unit->getPlayer()->getRace() != enemy.getRace()) {
```

Comment:
Might be better to check `enemy.getRace() == BWAPI::Races::Unknown`. Having multiple enemies in a game will be a bit more difficult, and would cause this to repeatedly be called once a new/different player unit is found.